packages/twenty-server/src/modules/messaging/message-import-manager/services/messaging-messages-import.service.ts

```diff
@@ -77,40 +77,42 @@ export class MessagingMessagesImportService {
         messageChannel.id,
       ]);
 
-      try {
-        connectedAccount.accessToken =
-          await this.refreshAccessTokenService.refreshAndSaveAccessToken(
-            connectedAccount,
-            workspaceId,
-          );
-      } catch (error) {
-        switch (error.code) {
-          case RefreshAccessTokenExceptionCode.REFRESH_ACCESS_TOKEN_FAILED:
-          case RefreshAccessTokenExceptionCode.REFRESH_TOKEN_NOT_FOUND:
-            await this.messagingTelemetryService.track({
-              eventName: `refresh_token.error.insufficient_permissions`,
+      if (connectedAccount.provider === 'google') {
```

Comment:
no need to refresh for microsoft?

Reply:
or for later?

Reply:
no need. using the client only required the refreshtoken. To make batch calls on gmail, we had to go through axios and therefore refresh the access token